src/server/system_services/pool_controllers.js

```diff
@@ -159,21 +118,22 @@ class ManagedStatefulSetPoolController extends PoolController {
 // have no control over the underlying storage
 // ----------------------------------------------
 class UnmanagedStatefulSetPoolController extends PoolController {
-    constructor(system_name, pool_name) {
-        super(system_name, pool_name);
-        this._statefulset_name = _get_statefulset_name(system_name, pool_name);
-    }
-
     // Just return the configuration to apply in order to create
     // the statefulset.
     async create(agent_count, agent_install_conf, agent_profile) {
+        const secret_k8s_conf = await _get_k8s_secret({
+            pool_name: this.pool_name,
+            agent_install_conf,
+        });
         const pool_k8s_conf = await _get_k8s_conf({
-            statefulset_name: this._statefulset_name,
+            pool_name: this.pool_name,
             agent_count,
-            agent_install_conf,
             ...agent_profile
         });
-        return yaml_utils.stringify(pool_k8s_conf);
+        const secret_yaml = yaml.stringify(secret_k8s_conf, 6, 2);
+        const backstore_yaml = yaml.stringify(pool_k8s_conf, 6, 2);
+
+        return `${secret_yaml}---\n${backstore_yaml}`;
```

Comment:
Please use yaml_utils for this,  it can handle stringifying into a multi document YAML.
```
return yaml_utils.stringify([secret_k8s_conf, pool_k8s_conf])
```

Reply:
it's a 2 document yaml - will reconsider if we get issues